packages/legend-query-builder/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -269,75 +320,346 @@ const QueryBuilderGridResultContextMenu = observer(
     );
   }),
 );
 
+const queryResultCellRenderer = observer(
+  (
+    params: DataGridCellRendererParams<{
+      resultState: QueryBuilderResultState;
+      cellExecutionResult: TDSExecutionResult;
+    }>,
+  ) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const resultState = (params as any).resultState as QueryBuilderResultState;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const cellExecutionResult = (params as any)
+      .cellExecutionResult as TDSExecutionResult;
+
+    const cellValue = params.value as string;
+
+    const columnName = params.column?.getColId() ?? '';
+
+    const findCoordinatesFromResultValue = (
+      colId: string,
+      rowNumber: number,
+    ): ResultCoordinate => {
+      const colIndex = cellExecutionResult.result.columns.findIndex(
+        (col) => col === colId,
+      );
+
+      return { rowIndex: rowNumber, colIndex: colIndex };
+    };
+
+    const currentCellCoordinates = findCoordinatesFromResultValue(
+      columnName,
+      params.rowIndex,
+    );
+
+    const cellInFilteredResults = resultState.filteredResults.some(
+      (result) =>
+        result.coordinates.colIndex === currentCellCoordinates.colIndex &&
+        result.coordinates.rowIndex === currentCellCoordinates.rowIndex,
+    );
+
+    const mouseDown: React.MouseEventHandler = (event) => {
+      event.preventDefault();
+
+      if (event.shiftKey) {
+        const coordinates = findCoordinatesFromResultValue(
+          columnName,
+          params.rowIndex,
+        );
+        const actualValue = resultState.findResultValueFromCoordinates([
+          coordinates.rowIndex,
+          coordinates.colIndex,
+        ]);
+        resultState.addFilteredResult({
+          value: actualValue,
+          columnName: columnName,
+          coordinates: coordinates,
+        });
+        return;
+      }
+
+      if (event.button === 0) {
+        resultState.setIsHoldingCells(true);
+        resultState.setFilteredResults([]);
+        const coordinates = findCoordinatesFromResultValue(
+          columnName,
+          params.rowIndex,
+        );
+        const actualValue = resultState.findResultValueFromCoordinates([
+          coordinates.rowIndex,
+          coordinates.colIndex,
+        ]);
+
+        const rowNode = params.api.getRowNode(params.rowIndex.toString());
+
+        if (rowNode) {
+          params.api.refreshCells({
+            force: true,
+            columns: [columnName],
+            rowNodes: [rowNode],
+          });
+        }
+        resultState.setFilteredResults([
+          {
+            value: actualValue,
+            columnName: columnName,
+            coordinates: coordinates,
+          },
+        ]);
+      }
+    };
+    const mouseUp: React.MouseEventHandler = (event) => {
+      resultState.setIsHoldingCells(false);
+    };
+
+    const mouseOver: React.MouseEventHandler = (event) => {
+      if (resultState.holdingCells) {
```

Comment:
I think we can make `holdingCells` or `isSelectingCells` a react `useState`, it's really just at the UI interaction layer, no need to put this in a state for much reasons